src/app.rs

```diff
@@ -1,23 +1,23 @@
+use crate::virtual_dom::{self, Attribute, Event, Html, Node};
+use std::cell::Ref;
+use std::fmt;
 use std::{cell::RefCell, rc::Rc};
-
-use crate::virtual_dom::{self, Attribute, Html, Node};
 use wasm_bindgen::{prelude::Closure, JsCast, JsValue};
-use web_sys::{Document, Element};
+use web_sys::console;
+use web_sys::{Document, Element, InputEvent};
 
-#[derive(Clone, Debug, Eq, PartialEq)]
 struct State<Model, Message>
 where
-    Model: Clone + Eq,
-    Message: Clone + Eq,
```

Comment:
I removed a lot of unnecessary constraints as they were making it hard to get things compiling.